param.go

```diff
@@ -146,28 +146,11 @@ func (pl paramList) Build(containerStore) (reflect.Value, error) {
 // to the underlying constructor.
 func (pl paramList) BuildList(c containerStore) ([]reflect.Value, error) {
 	args := make([]reflect.Value, len(pl.Params))
-	allContainers := c.storesToRoot()
 	for i, p := range pl.Params {
-		// iterate through the tree path of scopes.
-	containerLoop:
-		for _, c := range allContainers {
-			arg, err := p.Build(c)
-			if err == nil {
-				args[i] = arg
-				break containerLoop
-			}
-			// If argument has successfully been built, it's possible
-			// for these errors to occur in child scopes that don't
-			// contain the given parameter type. We can safely ignore
-			// these.
-			// If it's an error other than missing types/dependencies,
-			// this means some constructor returned an error that must
-			// be reported.
-			_, isErrMissingTypes := err.(errMissingTypes)
-			_, isErrMissingDeps := err.(errMissingDependencies)
-			if err != nil && !isErrMissingTypes && !isErrMissingDeps {
-				return nil, err
-			}
+		var err error
+		args[i], err = p.Build(c)
+		if err != nil {
```

Comment:
```suggestion
		if args[i], err := p.Build(c); err != nil {
```

Reply:
Can't do this because we're assigning to a slice element. `:=` only works if we create a new variable.